mirrord/operator/src/client.rs

```diff
@@ -748,23 +754,37 @@ impl OperatorApi<PreparedClientCert> {
             .map_err(KubeApiError::from)
             .map_err(OperatorApiError::CreateKubeClient)?;
 
-        let (tx, rx) = Self::connect_target(&client, &session).await?;
+        let (tx, rx) = Self::connect_target(&client, &session, layer_config).await?;
 
         Ok(OperatorSessionConnection { tx, rx, session })
     }
 
     /// Creates websocket connection to the operator target.
+    ///
+    /// - `layer_config`: Helps us include additional info about the target connection, like sending
+    ///   SQS setup.
     #[tracing::instrument(level = Level::TRACE, skip(client), err)]
     async fn connect_target(
         client: &Client,
         session: &OperatorSession,
+        layer_config: &LayerConfig,
     ) -> OperatorApiResult<(Sender<ClientMessage>, Receiver<DaemonMessage>)> {
         let request = Request::builder()
             .uri(&session.connect_url)
-            .header(SESSION_ID_HEADER, session.id.to_string())
+            .header(SESSION_ID_HEADER, session.id.to_string());
+
+        let request = if layer_config.feature.split_queues.is_set() {
+            request.header(
+                HeaderName::from_static(CLIENT_SQS_HEADER),
+                HeaderValue::from_bytes(&serde_json::to_vec(&layer_config.feature.split_queues)?)?,
+            )
+        } else {
+            request
+        };
+
+        let request = request
```

Comment:
Why change the way we send the SQS config?

Reply:
It's not changing it. This is being consumed on the operator side `connect_resource` because there we don't get any SQS info, only that we're connecting to a copy target. I've tried changing it so SQS would bleed over to `connect_resource`, but it seemed like I would need to introduce a bunch of changes (some potentially breaking due to CRDs?).

Just throwing the SQS stuff into a header seemed way simpler.

Reply:
@t4lz I think the review should start from the operator side.
Trusting user input to determine state is wrong - operator should have information about what current splitting is going on and shouldn't need to rely on user passing the information for it.
In general I think the SQS (and other splitting) status should be in another resource, that can be fetched as part of mirrord operator status (so the status will be built from more than one api call) but that's not a must.

Reply:
I suggest the following flow for SQS status:
1. Client queries mirrord operator's status.
2. mirrord operator determines SQS operator status via Kube API, either:
  - mirrord operator gets the status of all `MirrordWorkloadQueueRegistry` objects (nice if we want to group by workload), or
  - mirrord operator gets the status of all `MirrordSQSSession` objects, or
  - the SQS operator keeps its status in a CR, and the mirrord operator reads that status from there.
3. mirrord operator returns complete status (including SQS) to client.

What do you guys think?

Reply:
As I don't have a good understanding of what SQS is:

- Is the SQS something that lives longer than the user session that started it, like a copy target pod?
- To implement any of these suggestions, I'll need help getting a running SQS to test stuff.